x-pack/test/ui_capabilities/spaces_only/scenarios.ts

```diff
@@ -0,0 +1,202 @@
+}
+const DiscoverDisabledSpace: DiscoverDisabledSpace = {
+  id: 'discover_disabled_space',
+  name: 'discover_disabled_space',
+  disabledFeatures: ['discover'],
+};
+
+interface GisDisabledSpace extends Space {
+  id: 'gis_disabled_space';
+}
+const GisDisabledSpace: GisDisabledSpace = {
+  id: 'gis_disabled_space',
+  name: 'gis_disabled_space',
+  disabledFeatures: ['gis'],
+};
+
+interface GraphDisabledSpace extends Space {
+  id: 'graph_disabled_space';
+}
+const GraphDisabledSpace: GraphDisabledSpace = {
+  id: 'graph_disabled_space',
+  name: 'graph_disabled_space',
+  disabledFeatures: ['graph'],
+};
+
+interface InfrastructureDisabledSpace extends Space {
+  id: 'infrastructure_disabled_space';
+}
+const InfrastructureDisabledSpace: InfrastructureDisabledSpace = {
+  id: 'infrastructure_disabled_space',
+  name: 'infrastructure_disabled_space',
+  disabledFeatures: ['infrastructure'],
+};
+
+interface LogsDisabledSpace extends Space {
+  id: 'logs_disabled_space';
+}
+const LogsDisabledSpace: LogsDisabledSpace = {
+  id: 'logs_disabled_space',
+  name: 'logs_disabled_space',
+  disabledFeatures: ['logs'],
+};
+
+interface MlDisabledSpace extends Space {
+  id: 'ml_disabled_space';
+}
+const MlDisabledSpace: MlDisabledSpace = {
+  id: 'ml_disabled_space',
+  name: 'ml_disabled_space',
+  disabledFeatures: ['ml'],
+};
+
+interface MonitoringDisabledSpace extends Space {
+  id: 'monitoring_disabled_space';
+}
+const MonitoringDisabledSpace: MonitoringDisabledSpace = {
+  id: 'monitoring_disabled_space',
+  name: 'monitoring_disabled_space',
+  disabledFeatures: ['monitoring'],
+};
+
+interface TimelionDisabledSpace extends Space {
+  id: 'timelion_disabled_space';
+}
+const TimelionDisabledSpace: TimelionDisabledSpace = {
+  id: 'timelion_disabled_space',
+  name: 'timelion_disabled_space',
+  disabledFeatures: ['timelion'],
+};
+
+interface VisualizeDisabledSpace extends Space {
+  id: 'visualize_disabled_space';
+}
+const VisualizeDisabledSpace: VisualizeDisabledSpace = {
+  id: 'visualize_disabled_space',
+  name: 'visualize_disabled_space',
+  disabledFeatures: ['visualize'],
+};
+
+export type SpaceScenarios =
+  | EverythingSpace
+  | NothingSpace
+  | AdvancedSettingsDisabledSpace
+  | ApmDisabledSpace
+  | CanvasDisabledSpace
+  | DashboardDisabledSpace
+  | DevToolsDisabledSpace
+  | DiscoverDisabledSpace
+  | GisDisabledSpace
+  | GraphDisabledSpace
+  | InfrastructureDisabledSpace
+  | LogsDisabledSpace
+  | MlDisabledSpace
+  | MonitoringDisabledSpace
+  | TimelionDisabledSpace
+  | VisualizeDisabledSpace;
+export const SpaceScenarios: SpaceScenarios[] = [
+  EverythingSpace,
+  NothingSpace,
+  AdvancedSettingsDisabledSpace,
+  ApmDisabledSpace,
+  CanvasDisabledSpace,
+  DashboardDisabledSpace,
+  DevToolsDisabledSpace,
+  DiscoverDisabledSpace,
+  GisDisabledSpace,
+  GraphDisabledSpace,
+  InfrastructureDisabledSpace,
+  LogsDisabledSpace,
+  MlDisabledSpace,
+  MonitoringDisabledSpace,
+  TimelionDisabledSpace,
```

Comment:
I only noticed this because the test for the visualize app was incorrect, but I didn't have any test failures locally. I'm not sure if we have a way to guard against this from happening in the future though.

```suggestion
  TimelionDisabledSpace,
  VisualizeDisabledSpace,
```

Reply:
Great catch, thanks! I wasn't able to find a way to guard against this for the future, I fought to get a single "list", but wasn't able to find a way using the typescript type system. We need the `type` for the exhaustive switch, but we also need to be able to iterate over the options...